File: lib/tmdb_api/model/gender.dart
class Gender {
  final int _id;

  const Gender(this._id);

  static const unknown = Gender(0);
  static const female = Gender(1);
  static const male = Gender(2);

  static const values = [unknown, female, male];

  factory Gender.valueOf(int json) {
    if (json ==null) return Gender.unknown;
    return values.firstWhere((v) => json == v._id) ?? Gender.unknown;
  }

  factory Gender.fromJson(Map<String, dynamic> json) {
    return Gender.valueOf(json['gender']);
  }
}

File: kmp/model/src/main/java/com/tikal/tmdb/json/model/Gender.kt
package com.tikal.tmdb.json.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
enum class Gender {
    @SerialName("0")
    unknown,
    @SerialName("1")
    female,
    @SerialName("2")
    male
}